illumina.py

```diff
@@ -182,22 +195,34 @@ def main_illumina_demux(args):
         picardOpts['num_processors'] = args.threads
     if not picardOpts.get('sequencing_center') and illumina.get_RunInfo():
         picardOpts['sequencing_center'] = illumina.get_RunInfo().get_machine()
-    tools.picard.IlluminaBasecallsToSamTool().execute(
-        illumina.get_BCLdir(),
-        barcodes_tmpdir,
-        flowcell,
-        args.lane,
-        basecalls_input,
-        picardOptions=picardOpts,
-        JVMmemory=args.JVMmemory)
+
+    if multiplexed_samples:
+        tools.picard.IlluminaBasecallsToSamTool().execute(
+            illumina.get_BCLdir(),
+            barcodes_tmpdir,
+            flowcell,
+            args.lane,
+            basecalls_input,
+            picardOptions=picardOpts,
+            JVMmemory=args.JVMmemory)
+    else:
+        tools.picard.IlluminaBasecallsToSamTool().execute_single_sample(
+            illumina.get_BCLdir(),
+            os.path.join(args.outDir,flowcell+".bam"),
```

Comment:
If we break out a separate parser for simplex bcl-to-bam, it should allow the user to specify an output file rather than an output directory.